piaf.opam

```diff
@@ -7,6 +7,7 @@ bug-reports: "https://github.com/anmonteiro/piaf/issues"
 dev-repo: "git+https://github.com/anmonteiro/piaf.git"
 doc: "https://anmonteiro.github.io/piaf/"
 build: [
+  ["git" "submodule" "update" "--init" "--recursive"]
```

Comment:
`dune-release` pulls submodules when submitting to opam, so this is unnecessary.

I'd be happy to incorporate instructions for cloning the submodules in either the README or a CONTRIBUTING.md file.

Reply:
This was added as part of the devcontainer build (see below). I can take it out and move to script if you are still interested in this PR